src/components/Api.js

```diff
@@ -0,0 +1,105 @@
+export default class Api {
+  constructor(baseUrl, headers) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  loadUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject("Failed to fetch initial card").catch((err) => {
+        return Promise.reject(`Error ${err}`);
+      });
+    });
+  }
+
+  editingProfile(profileInfo) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify(profileInfo),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject("Failed to fetch edit profile");
+      })
+      .catch((err) => {
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.


And you catch and rethrow the error again without changing, so this block is redundant here